apps/src/StudioApp.js

```diff
@@ -733,61 +728,30 @@ StudioApp.prototype.loadBlocks = function(blocksXml) {
 *    block position.
 */
 StudioApp.prototype.arrangeBlockPosition = function(startBlocks, arrangement) {
-  var type, arrangeX, arrangeY;
+
+  var type, x, xmlChild;
+
   var xml = parseXmlElement(startBlocks);
-  var xmlChildNodes = this.sortBlocksByVisibility(xml.childNodes);
-  var numberOfPlacedBlocks = 0;
-  for (var x = 0, xmlChild; xmlChildNodes && x < xmlChildNodes.length; x++) {
+
+  var xmlChildNodes = xml.childNodes || [];
+  arrangement = arrangement || {};
+
+  for (x = 0; x < xmlChildNodes.length; x++) {
```

Comment:
Any reason you've manually hoisted the `var x` declaration out of this for loop?  It doesn't get used anywhere outside of the loop - I know the scope is the same in the end, but for a simple index variable we usually leave it in.  Also, I'd prefer to see `i` over `x` for the index variable here - with `x` I did a double-take wondering if it was something other than an index.